File: playfunction/playfunction/report/order_history/order_history.py
# Copyright (c) 2013, Indictrans and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns = [
		{
			'fieldname': 'name',
			'label': 'Order No.',
			'fieldtype': 'Link',
			'options': 'Sales Order'
		},
		{
			'fieldname': 'customer',
			'label': 'Customer',
			'fieldtype': 'Link',
			'options': 'Customer'
		},
		{
			'fieldname': 'transaction_date',
			'fieldtype': 'Date',
			'label': 'Order Date'
		},
		{
			'fieldname': 'grand_total',
			'fieldtype': 'Currency',
			'label': 'Order Value'
		},
		{
			'fieldname': 'status',
			'fieldtype': 'Data',
			'label': 'Order Status'
		},
		{
			'fieldname': 'shipping_status',
			'fieldtype': 'Data',
			'label': 'Shipping Status'
		},
		{
			'fieldname': 'payment_status',
			'fieldtype': 'Data',
			'label': 'Payment Status'
		}
	]
	query = """
			select name, customer, transaction_date, grand_total, status, 
			'-' as shipping_status, '-' as payment_status
			from `tabSales Order`
		"""
	cond = " where 1=1"
	if filters.get('from_date') and filters.get('to_date'):
		cond += " and transaction_date >= '{0}' and transaction_date <= '{1}'".format(
			filters.get("from_date"), filters.get("to_date"))
	if filters.get("status"):
		cond += " and status = '{}'".format(filters.get("status"))
	data = frappe.db.sql(query + cond)
	return columns, data

File: playfunction/playfunction/report/order_history/order_history.js
// Copyright (c) 2016, Indictrans and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Order History"] = {
	"filters": [
		{
			"fieldname": "from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			'reqd': 1,
			"default": frappe.datetime.add_days(frappe.datetime.nowdate(), -30)
		},
		{
			"fieldname": "to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			'reqd': 1,
			"default": frappe.datetime.nowdate()
		},
		{
			"fieldname": "grand_total",
			"label": "Order Value",
			"fieldtype": "Currency"
		},
		{
			"fieldname": "status",
			"label": "Order Status",
			"fieldtype": "Select",
			"options": "\nDraft\nTo Deliver and Bill\nTo Bill\nTo Deliver\nCompleted\nCancelled\nClosed"
		},
		{
			"fieldname": "shipping_status",
			"label": "Shippin Status",
			"fieldtype": "Select",
			"options": "\nFully Shipped\nNot Shipped\nPartially Shipped",
		},
		{
			"fieldname": "payment_status",
			"label": "Payment Status",
			"fieldtype": "Select",
			"options": "\nDue\nInvoice Sent\nPaid"
		},
	]
}
